File: app/src/main/java/com/venditti/placares/adapter/ScoreboardAdapter.java
package com.venditti.placares.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import com.venditti.placares.R;
import com.venditti.placares.model.Players;
import com.venditti.placares.model.Points;

import java.util.HashMap;
import java.util.List;

public class ScoreboardAdapter extends RecyclerView.Adapter<ScoreboardAdapter.MyViewHolder>{
    private List<Players> listJogadores;
    private ScoreboardPointsAdapter adapter;
    private Context context;

    public ScoreboardAdapter(Context context,List<Players> listJogadores) {
        this.context = context;
        this.listJogadores = listJogadores;
    }

    @NonNull
    @Override
    public MyViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View scoreboard = LayoutInflater.from(parent.getContext()).inflate(R.layout.player_scoreboard_layout, parent, false);
        return new MyViewHolder(scoreboard);
    }

    @Override
    public void onBindViewHolder(@NonNull MyViewHolder holder, int position) {
        holder.textPlayerScoreboard.setText(listJogadores.get(holder.getAdapterPosition()).getName());
        holder.textTotalGeral.setText(listJogadores.get(holder.getAdapterPosition()).getTotal().toString());
        HashMap<String, Points> points = listJogadores.get(holder.getAdapterPosition()).getPoints();

        if(!points.isEmpty()){
            adapter = new ScoreboardPointsAdapter(points, points.keySet().toString());
            //Define Layout Categoria
            RecyclerView.LayoutManager layoutManager = new LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false);
            holder.recyclerPoints.setLayoutManager(layoutManager);
            holder.recyclerPoints.setHasFixedSize(true);
            holder.recyclerPoints.setAdapter(adapter);
        }
    }

    @Override
    public int getItemCount() {
        return listJogadores.size();
    }

    public static class MyViewHolder extends RecyclerView.ViewHolder {

        TextView textPlayerScoreboard, textTotalGeral;
        RecyclerView recyclerPoints;
        public MyViewHolder(@NonNull View itemView) {
            super(itemView);
            textTotalGeral = itemView.findViewById(R.id.textTotalGeral);
            textPlayerScoreboard = itemView.findViewById(R.id.textPlayerScoreboard);
            recyclerPoints = itemView.findViewById(R.id.recyclerPoints);
        }
    }
}

File: app/src/main/java/com/venditti/placares/adapter/ScoreboardAdapter.kt
package com.venditti.placares.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.venditti.placares.R
import com.venditti.placares.model.Players

class ScoreboardAdapter(private val context: Context?, private val listJogadores: List<Players>?) : RecyclerView.Adapter<ScoreboardAdapter.MyViewHolder>() {
    private var adapter: ScoreboardPointsAdapter? = null
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): MyViewHolder {
        val scoreboard = LayoutInflater.from(parent.context).inflate(R.layout.player_scoreboard_layout, parent, false)
        return MyViewHolder(scoreboard)
    }

    override fun onBindViewHolder(holder: MyViewHolder, position: Int) {
        holder.textPlayerScoreboard.text = listJogadores!![holder.adapterPosition].name
        holder.textTotalGeral.text = listJogadores[holder.adapterPosition].total.toString()
        val points = listJogadores[holder.adapterPosition].points
        if (!points!!.isEmpty()) {
            adapter = ScoreboardPointsAdapter(points, points.keys.toString())
            //Define Layout Categoria
            val layoutManager: RecyclerView.LayoutManager = LinearLayoutManager(context, LinearLayoutManager.VERTICAL, false)
            holder.recyclerPoints.layoutManager = layoutManager
            holder.recyclerPoints.setHasFixedSize(true)
            holder.recyclerPoints.adapter = adapter
        }
    }

    override fun getItemCount(): Int {
        return listJogadores!!.size
    }

    class MyViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var textPlayerScoreboard: TextView
        var textTotalGeral: TextView
        var recyclerPoints: RecyclerView

        init {
            textTotalGeral = itemView.findViewById(R.id.textTotalGeral)
            textPlayerScoreboard = itemView.findViewById(R.id.textPlayerScoreboard)
            recyclerPoints = itemView.findViewById(R.id.recyclerPoints)
        }
    }
}